### PR TITLE
Feature/gh 5478

### LINKED ISSUE #5478: Add "return requests list" tab in order list page

### PR DESCRIPTION
**Pull request by WeizhengSap**
close #5478

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/myaccount/order/order-history/order-history.component.html [hunk 1]
```diff
@@ -1,9 +1,11 @@
 <ng-container *ngIf="orders$ | async as orders">
   <div class="container">
     <!-- HEADER -->
+    <!--
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/cms-components/myaccount/order/order-history/order-history.component.html]: I just comment it out. `Header` should be displayed when the feature is disabled. Toggle feature will be handled in #4659

File: projects/storefrontlib/src/cms-components/myaccount/order/order-return-requests/order-return-requests.component.html [hunk 2]
```diff
@@ -0,0 +1,118 @@
+<ng-container *ngIf="returnRequests$ | async as returnRequests">
```
Comment (LTiger14) [inline projects/storefrontlib/src/cms-components/myaccount/order/order-return-requests/order-return-requests.component.html]: This is a global question, so `order-return-requests` is the list. Can't it be call `order-return-request-list`? 

Also, should we add a list item component like we have in cart and other lists

File: projects/core/src/user/store/effects/order-return-request.effect.ts [hunk 3]
```diff
@@ -66,6 +69,17 @@ export class OrderReturnRequestEffect {
     })
   );
 
+  @Effect()
+  resetReturnRequestList$: Observable<Action> = this.actions$.pipe(
```
[the remark this answers is not in the document]
Comment (WeizhengSap) [reply projects/core/src/user/store/effects/order-return-request.effect.ts]: added

File: projects/storefrontlib/src/cms-components/myaccount/order/order-return-requests/order-return-requests.component.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+   * TabParagraphContainerComponent. This can be read from TabParagraphContainer.
+   */
+  tabTitleParam$: Observable<number> = this.returnRequests$.pipe(
+    map(returnRequests => returnRequests.pagination.totalResults),
+    filter(totalResults => totalResults !== undefined),
+    take(1)
+  );
+
+  ngOnDestroy(): void {
+    this.returnRequestService.clearOrderReturnRequestList();
```
[the remark this answers is not in the document]
Comment (WeizhengSap) [reply projects/storefrontlib/src/cms-components/myaccount/order/order-return-requests/order-return-requests.component.ts]: this component is mimic the "OrderHistoryComponent". So, when leave the "return list" page, the data is cleared. Next time, when enter in, then the new data will be loaded.

File: projects/storefrontlib/src/cms-components/myaccount/order/order-return-requests/order-return-requests.component.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    })
+  );
+
+  /**
+   * When "Order Return" feature is enabled, this component becomes one tab in
+   * TabParagraphContainerComponent. This can be read from TabParagraphContainer.
+   */
+  tabTitleParam$: Observable<number> = this.returnRequests$.pipe(
+    map(returnRequests => returnRequests.pagination.totalResults),
+    filter(totalResults => totalResults !== undefined),
```
[the remark this answers is not in the document]
Comment (WeizhengSap) [reply projects/storefrontlib/src/cms-components/myaccount/order/order-return-requests/order-return-requests.component.ts]: the initial value of pagination is `{}`. so, `pagination.totalResults` can be `undefined`, right?

File: projects/storefrontlib/src/cms-components/myaccount/order/order-return-requests/order-return-requests.component.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+  getSortLabels(): Observable<{ byDate: string; byRMA: string }> {
+    return combineLatest([
+      this.translation.translate('sorting.date'),
+      this.translation.translate('sorting.rma'),
+    ]).pipe(
+      map(([textByDate, textByRma]) => {
+        return {
+          byDate: textByDate,
+          byRMA: textByRma,
```
[the remark this answers is not in the document]
Comment (WeizhengSap) [reply projects/storefrontlib/src/cms-components/myaccount/order/order-return-requests/order-return-requests.component.ts]: Return Merchandise Authorization (RMA).

File: projects/storefrontlib/src/cms-components/myaccount/order/order-return-requests/order-return-requests.component.html (hunk 2 above)
Comment (WeizhengSap) [reply projects/storefrontlib/src/cms-components/myaccount/order/order-return-requests/order-return-requests.component.html]: yes!, I will rename them.

File: projects/storefrontlib/src/cms-components/myaccount/order/order-return-requests/order-return-requests.component.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (LTiger14) [reply projects/storefrontlib/src/cms-components/myaccount/order/order-return-requests/order-return-requests.component.ts]: Yes true

File: projects/storefrontlib/src/cms-components/myaccount/order/order-return-requests/order-return-requests.component.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (LTiger14) [reply projects/storefrontlib/src/cms-components/myaccount/order/order-return-requests/order-return-requests.component.ts]: All these TLAs

**Pull request merged**